plugins/enthusiast-common/enthusiast_common/repositories/django.py

```diff
@@ -0,0 +1,86 @@
+from typing import TypeVar, Type, Optional, Generic
+
+from django.db import models
+from django.db.models import QuerySet
+from pgvector.django import CosineDistance
+
+from .base import (
+    BaseRepository,
+    BaseMessageRepository,
+    BaseConversationRepository,
+    BaseUserRepository,
+    BaseDataSetRepository,
+    BaseDocumentChunkRepository,
+    BaseProductRepository,
+)
+
+T = TypeVar("T", bound=models.Model)
+U = TypeVar("U", bound=models.Model)
+
+
+class BaseDjangoRepository(BaseRepository[T]):
```

Comment:
I'd put these classes inside the server too - so that we don't have to distribute the implementation in a package that's mainly for interfaces that allow creating plugins

Reply:
In general I agree, but there was two things that convinced me to place in in common package
1. During our contribution to Enthusiast more code in common will probably reduce number of conflicts for users
2. I was thinking of it as user may want to extend functionality this in future developed packages